### ISSUE TITLE
请问怎么去改成生成视频的长度？

### ISSUE THREAD

**Issue by gangxu822** [DINGYANB/MTVCrafter#11]

**Comment by DINGYANB** [comment]
You need to modify the number of frames for generation (e.g. 72, 98, 128) and disable the motion classifier free-guidance, since our unconditional motion tokens were specifically trained for 49-frame videos.

Maybe you can try expanding the unconditional motion tokens to longer frames, but we haven’t tested that. It may potentially lead to a drop in generation quality.

**Comment by gangxu822** [comment]
> You need to modify the number of frames for generation (e.g. 72, 98, 128) and disable the motion classifier free-guidance, since our unconditional motion tokens were specifically trained for 49-frame videos.
> 
> Maybe you can try expanding the unconditional motion tokens to longer frames, but we haven’t tested that. It may potentially lead to a drop in generation quality.

多谢了，试了效果不太好，再去试试unianimate-dit animate-x

**Comment by blahole** [comment]
> You need to modify the number of frames for generation (e.g. 72, 98, 128) and disable the motion classifier free-guidance, since our unconditional motion tokens were specifically trained for 49-frame videos.
> 
> Maybe you can try expanding the unconditional motion tokens to longer frames, but we haven’t tested that. It may potentially lead to a drop in generation quality.

How to disable the motion classifier free-guidance?

**Comment by blahole** [comment]
> > You need to modify the number of frames for generation (e.g. 72, 98, 128) and disable the motion classifier free-guidance, since our unconditional motion tokens were specifically trained for 49-frame videos.
> > Maybe you can try expanding the unconditional motion tokens to longer frames, but we haven’t tested that. It may potentially lead to a drop in generation quality.
> 
> How to disable the motion classifier free-guidance?

The guidance_scale para~

**Comment by DINGYANB** [comment]
> > > You need to modify the number of frames for generation (e.g. 72, 98, 128) and disable the motion classifier free-guidance, since our unconditional motion tokens were specifically trained for 49-frame videos.
> > > Maybe you can try expanding the unconditional motion tokens to longer frames, but we haven’t tested that. It may potentially lead to a drop in generation quality.
> > 
> > 
> > How to disable the motion classifier free-guidance?
> 
> The guidance_scale para~

You can try modifying the Line 636 of `pipeline_mtvcrafter.py` to `motion_embeds = torch.cat([motion_embeds, motion_embeds], dim=0)  
`
